posydon/binary_evol/SN/step_SN.py

```diff
@@ -503,22 +503,56 @@ def collapse_star(self, star):
                 elif getattr(star, MODEL_NAME_SEL) is None:
                     # NOTE: this option is needed to do the collapse
                     # for stars evolved with the step_detached or 
-                    # steo_disrupted.
+                    # step_disrupted.
                     # allow to continue with the collapse with profile
                     # or core masses
                     warnings.warn(f'{MODEL_NAME_SEL}: The collapsed star '
                                      'was not interpolated! If use_profiles '
                                      'or use_core_masses is set to True, '
                                      'continue with the collapse.')                 
                 else:
+                    # store some properties of the star object
+                    # to be used for collapse verification
+                    pre_SN_c_core_mass = star.c_core_mass
+                    pre_SN_he_core_mass = star.he_core_mass
+                    pre_SN_mass = star.mass
+                    
                     MODEL_properties = getattr(star, MODEL_NAME_SEL)
                     for key, value in MODEL_properties.items():
                         setattr(star, key, value)
-                
+                    
+
                     for key in STARPROPERTIES:
                         if key not in ["state", "mass", "spin",
                                         "m_disk_accreted ", "m_disk_radiated"]:
                             setattr(star, key, None)
+                            
+                    # check if SN_type is not None
+                    print('state:',star.state)
+                    print('type:',star.SN_type)  
+                    # check if SN_type matches the predicted CO
+                    # ie WD is no SN
+                    if star.state == 'WD' and star.SN_type != "WD":
+                        star.SN_type = 'WD'
+                    elif (star.state == "NS" or star.state == "BH") and \
+                         (star.SN_type != 'ECSN' or star.SN_type != 'CCSN'):
```

Comment:
I'm wondering, whether it would be better to check, that SN_type contains "SN". This would allow to be more general and include things like "PPISN".

Reply:
Is `PPISN` an output of the SN_type interpolator?

I think this will lead into issues with `PISN`, since then the star state should be `PISN` as well. Thus, BH+PISN would be a wrong prediction.

Reply:
Here it might be the case, that the code isn't up to date with physics.

The SN_type `PPISN` should lead to a CO_type(star.state) `BH`, while SN_type `PISN` should lead to CO_type(star.state) `massless_remnant`.